services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/MetersSpec.groovy

```diff
@@ -55,87 +57,96 @@ class MetersSpec extends BaseSpecification {
 
         then: "Delete operation should be successful"
         deleteResult.deleted
+        !northbound.getAllMeters(sw.dpId).meterEntries.find { it.meterId == meterToDelete }
 
         and: "Delete the flow"
         flowHelper.deleteFlow(flow.id)
 
         and: "Check if no excessive meters are installed on the switch"
         defaultMeters.meterEntries == northbound.getAllMeters(sw.dpId).meterEntries
+
+        where:
+        switchType   | switches
+        "Centec"     | getCentecSwitches()
+        "non-Centec" | getNonCentecSwitches()
     }
 
     @Unroll
     def "Unable to delete a meter with invalid ID=#meterId on a #switchType switch"() {
+        assumeTrue("Unable to find required switches in topology", switches as boolean)
+
         when: "Try to delete meter with invalid ID"
-        northbound.deleteMeter(sw.getDpId(), -1)
+        northbound.deleteMeter(switches[0].dpId, meterId)
 
         then: "Got BadRequest because meter ID is invalid"
         def exc = thrown(HttpClientErrorException)
         exc.rawStatusCode == 400
 
         where:
-        meterId | sw                        | switchType
-        -1      | getNonCentecSwitches()[0] | "non-Centec"
-        0       | getCentecSwitches()[0]    | "Centec"
-        -1      | getCentecSwitches()[0]    | "non-Centec"
-        0       | getNonCentecSwitches()[0] | "Centec"
+        meterId | switches               | switchType
+        -1      | getNonCentecSwitches() | "non-Centec"
```

Comment:
We can leave previous implementation with `sw` and add the assumption directly to the `getCertainSwitches` method. This is not that important, but will allow to get rid of duplicate assumptions in every test

Reply:
Unfortunately, we cannot. If the first test is skipped, all next iterations will be skipped as well. Probably, this behavior is introduced by the 'assumeTrue' method. My guess is that '
getCertainSwitches' method is called the very first, the condition is not satisfied and we skip the test not reaching Spock.